lib/MultiplyLink.sc

```diff
@@ -0,0 +1,18 @@
+/* 
+ * MultiplyLink: multiplies a signal by a provided factor.
+ *
+ * If the input signal and the factor are arrays of the same dimension, then
+ * multichannel expansion will be invoked and the factor array can be thought
+ * of as an array of weights.
+ */
+MultiplyLink : Link {
+    var<> factor;
+
+	*new { | factor=1.0 |
+		^super.new().factor_(factor);
+	}
+
+    transform { |input|
+        ^MulAdd(input, factor, 0)
```

Comment:
Did some local testing on this using some scratch SC work:

```
s.boot;

(
o = s.options;
o.numInputBusChannels = 3;
o.numOutputBusChannels = 6;
s.reboot;
s.meter;
)

(
SynthDef("test", {
	var signal;

        // signal = SinOsc.ar(174);
        // signal = [ SinOsc.ar(174) ];
	signal = [
		[SinOsc.ar(174), SinOsc.ar(220)],
		[SinOsc.ar(174), SinOsc.ar(261)],
		[SinOsc.ar(440), SinOsc.ar(440)],
	];
	signal = MulAdd(signal, 0.1);
	signal = MulAdd(signal, [0.85, 0.35, 0.1]);
	signal.postln;

	// Prints:
	// [ [ a BinaryOpUGen, a BinaryOpUGen ], [ a BinaryOpUGen, a BinaryOpUGen ], [ a BinaryOpUGen, a BinaryOpUGen ] ]

	// Since signal is an array of arrays, each output channel is the corresponding "inner" array squashed via sum
	// down to one channel
	Out.ar(0, signal);
}).send(s);

Synth("test");
)

s.freeAll;
s.reboot;
```

With running the synth, the meter does seem to indicate that if `input` is an array of array of signals and `factor` is an array of multipliers, then they do get multiplied element-wise correctly: 
![Screen Shot 2021-08-31 at 3 29 05 PM](https://user-images.githubusercontent.com/26987971/131584003-fb01b52f-9a7d-4d2b-ae2f-c56313081a2d.png)

(Only three output channels are displayed here because `signal` is an array of arrays, and each output channel produced by `Out` is generated by squashing each "inner" array. I can hear the F 174hz and A 220hz in the left ear, the F 174hz and C 261hz in the right ear. The A 440 hz is not audible with my 2 channel headset).

I also tested this with `signal = SinOsc.ar(174); ` and `signal = [ SinOsc.ar(174) ];` with the same meter output. I think this code does do what I expect it to do, but maybe your testing had a different setup? Or maybe there could be another problem elsewhere.